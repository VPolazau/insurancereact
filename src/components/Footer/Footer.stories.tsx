import { Story, Meta } from '@storybook/react';

import { Footer } from './Footer';

export default {
    title: 'UI/Footer',
    component: Footer,
} as Meta;

const Template: Story = (args) => <Footer {...args} />;

export const Default = Template.bind({});
Default.args = {};
