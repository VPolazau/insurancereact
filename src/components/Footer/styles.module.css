.footer {
  background-color: #EEFFF8;
  min-height: 286px;
  width: 100%;
  display: flex;
  justify-content: center;
  flex: 0 0 auto;
}

.logo {
  height: 46px;
  width: 150px;
}

.logoButton {
  line-height: 0;
}

.logoContainer {
  display: flex;
  flex-direction: column;
  gap: 15px;
}

.container {
  margin: 100px auto 106px;
  padding: 0 24px;
  /* ширина по макету + 2 падинга */
  max-width: calc(1218px + 48px);
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.infoContainer {
  display: flex;
  margin-right: 140px;
  gap: 112px;
}

.textContainer {
  max-width: 200px;
  display: flex;
  flex-direction: column;
  gap: 4px;
  padding-top: 12px;
}

.subtext {
  color: #646464;
}

@media screen and (max-width: 1150px) {
  .footer {
    min-height: 227px;  
  }

  .container {
    margin: 67px auto 0;
    align-items: flex-start;
  }

  .logoContainer {
    gap: 18px;
  }

  .infoContainer {
    margin-right: 199px;
    gap: 44px;
  }

  .textContainer {
    gap: 5px;
  }
}

@media screen and (max-width: 950px) {
  .infoContainer {
    margin-right: 83px;
    gap: 44px;
  }
}

@media (max-width: 650px) {
  .footer {
    height: 230px;
    justify-content: start;
  }

  .container {
    width: 100%;
    padding: 0;
    margin: 43px 30px 56px 16px;
  }

  .logoContainer {
    gap: 27px;
  }

  .infoContainer {
    flex-direction: column;
    justify-content: start;
    margin: 0;
    gap: 18px;
  }

  .textContainer {
    gap: 4px;
  }
}

@media (max-width: 360px) {
  .footer {
    height: 218px;
    justify-content: start;
  }

  .container {
    width: 100%;
    padding: 0;
    margin: 29px 24px 42px 16px;
    gap: 18px;
  }

  .logoContainer {
    gap: 11px;
    flex-basis: 64%;
  }

  .infoContainer {
    flex-direction: column;
    justify-content: start;
    margin: 0;
    gap: 16px;
  }

  .textContainer {
    gap: 4px;
  }
}