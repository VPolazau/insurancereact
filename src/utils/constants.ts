import boy from '../assets/img/boy.svg';
import car from '../assets/img/car.svg';

import { INotificationMessage } from '../redux/slices/user/types';

export interface Slides {
    title: string;
    subtitle: string;
    image?: string;
}

export const SLIDES: Slides[] = [
    {
        title: 'Защитите свою квартиру от&nbsp;неожиданных ситуаций',
        subtitle: 'получите страховку не&nbsp;выходя из&nbsp;дома',
        image: boy,
    },
    {
        title: 'Защитите свой автомобиль с&nbsp;нашей страховкой',
        subtitle: 'надежное покрытие от&nbsp;непредвиденных&nbsp;рисков!',
        image: car,
    },
];

export const CARDS = [
    {
        title: 'Широкий спектр услуг',
        description: 'Мы&nbsp;заботимся о&nbsp;Вашей безопасности',
    },
    {
        title: 'Чат поддержки',
        description: 'Получайте быстро и&nbsp;удобно ответы на&nbsp;вопросы',
    },
    {
        title: 'Личный кабинет',
        description: 'Управляйте своей&nbsp;страховкой легко и&nbsp;удобно',
    },
    {
        title: 'Выгодное предложение',
        description: 'Страховая сумма больше, а&nbsp;взнос&nbsp;меньше',
    },
];

export const PLACEHOLDERS = {
    policies: {
        title: 'У&nbsp;Вас пока нет заявок на&nbsp;страхование',
        subTitle:
            'В&nbsp;данном разделе&nbsp;Вы можете найти всю важную информацию, связанную с&nbsp;Вашими заявками на&nbsp;страхование и&nbsp;их&nbsp;статусами',
        buttonText: 'Оформить',
    },
    requests: {
        title: 'У&nbsp;Вас пока нет активных страховых полисов',
        subTitle:
            'Оформите страховой полис, и&nbsp;получите быстрый и&nbsp;удобный доступ ко&nbsp;всей информации о&nbsp;своих полисах в&nbsp;любое удобное для Вас время',
        buttonText: 'Оформить',
    },
    notifications: {
        title: 'У&nbsp;Вас пока нет уведомлений',
        subTitle:
            'В&nbsp;данном разделе&nbsp;Вы можете найти всю важную информацию, связанную с&nbsp;Вашими страховыми полисами, изменениями в&nbsp;Вашем профиле и&nbsp;новостями компании',
        buttonText: 'На главную',
    },
};

// TODO поменять все /in-progress на реальные страницы
export const NAVIGATION_DROP_LINKS = [
    {
        mainLinkPath: '/in-progress',
        mainLinkValue: 'Авто',
        dropNavLinks: [
            { dropLinkPath: '/in-progress', dropLinkValue: 'КАСКО' },
            { dropLinkPath: '/in-progress', dropLinkValue: 'ОСАГО' },
        ],
    },
    {
        mainLinkPath: '/in-progress',
        mainLinkValue: 'Имущество',
        dropNavLinks: [
            { dropLinkPath: '/property/apartment', dropLinkValue: 'Квартира' },
            { dropLinkPath: '/in-progress', dropLinkValue: 'Дом' },
        ],
    },
    {
        mainLinkPath: '/in-progress',
        mainLinkValue: 'Здоровье',
        dropNavLinks: [
            { dropLinkPath: '/in-progress', dropLinkValue: 'ДМС' },
            { dropLinkPath: '/in-progress', dropLinkValue: 'Страхование жизни' },
        ],
    },
    {
        mainLinkPath: '/in-progress',
        mainLinkValue: 'Отдых',
        dropNavLinks: [{ dropLinkPath: '/in-progress', dropLinkValue: 'Страхование путешественников' }],
    },
];

// TODO currentUser это будет пользователь полученный с бэка. "Выход" добавить роут на страницу авторизации
export const USER_DROP_LINKS = {
    currentUser: 'Максим Максимов',
    dropNavLinks: [
        { dropLinkPath: '/account', dropLinkValue: 'Личный кабинет' },
        { dropLinkPath: '/account/policies', dropLinkValue: 'Полисы' },
        { dropLinkPath: '/account/profile', dropLinkValue: 'Мой профиль' },
        { dropLinkPath: '/account/settings', dropLinkValue: 'Настройки' },
        { dropLinkPath: '/', dropLinkValue: 'Выход' },
    ],
};

export const PHONE_NUMBER: string = '+7 (495) 177 - 37 - 37';
export const MAIN_OFFICE_ADDRESS: string = 'г. Москва. Одесская улица 10, 5 этаж, офис 13';
export const COPYRIGHT = '&copy;&nbsp;2023 &nbsp;&laquo;My&nbsp;Insurance&raquo;';
export const IN_PROGRESS =
    'Уважаемые посетители, извините за&nbsp;неудобства, но&nbsp;наш сайт находится в&nbsp;активной разработке.';
export const EMAIL_CONFIRMED = 'Ваш email подтвержден!';

// TODO имитация карточек страховых полисов
export const INSURANCE_CARDS = [
    {
        id: '1',
        property: 'Квартира',
        type: 'Имущество',
        address: 'г. Москва, ул. Зеленоградская, д.102, к.5, кв 135',
        policyNumber: 'WIOO12357909',
        policyValidity: '01.01.2023-01.01.2024',
        policyCost: '17 000 ₽',
        paymentAmount: '8 500 ₽',
        paymentFrequency: 'Раз в 6 месяцев',
    },
    // Дополнительные объекты карточек полисов
];

// TODO имитация списка страховых заявок
export const REQUESTS = [
    {
        id: 1,
        type: 'Страхование квартиры',
        number: 2536324,
        created: '02.11.2022',
        status: 'approved',
        review: '2 дня',
        response: true,
        extension: true,
    },
    {
        id: 2,
        type: 'Страхование дома',
        number: 2834324,
        created: '02.11.2022',
        status: 'rejected',
        review: '2 дня',
        response: true,
    },
    {
        id: 3,
        type: 'КАСКО',
        number: 3534324,
        created: '02.11.2022',
        status: 'pending',
        review: '2 дня',
        response: false,
    },
];

// Имитация уведомлений в модалке
export const NOTIFICATIONS: INotificationMessage[] = [
    {
        id: '1',
        type: 'Страхование квартиры',
        title: 'Ваша заявка на «Страхование квартиры» одобрена',
        date: '04.11.2022',
        isRead: false,
        message:
            'Мы рады сообщить Вам, что Ваша заявка на страхование квартиры была рассмотрена и одобрена. Благодарим за проявленный интерес к нашей компании и надеемся на долгосрочное взаимодействие.',
    },
    {
        id: '2',
        type: 'Страхование дома',
        title: 'Ваша заявка на «Страхование дома» отклонена',
        date: '05.09.2022',
        isRead: false,
        message:
            'Уважаемый Константин Александрович! К сожалению, мы не можем удовлетворить вашу заявку на страхование дома...',
    },
    {
        id: '3',
        type: 'Смена пароля',
        title: 'Ваш пароль успешно изменен',
        date: '05.09.2022',
        isRead: true,
        message: 'Здравствуйте! Мы рады сообщить Вам, что пароль для вашей учетной записи был успешно изменен',
    },
];

export const RadioBtns = [
    { value: 'on', title: 'Вкл' },
    { value: 'off', title: 'Выкл' },
];

export const CONFIRM_POPUP_TITLES = {
    request: 'Вы действительно хотите отозвать&nbsp;заявку?',
    notification: 'Вы действительно хотите удалить&nbsp;уведомление?',
};

// TODO контент модалки ответа
export const RESPONSE_POPUP_CONTENT = {
    commonText: {
        title: 'Добрый день!',
        signature: 'С&nbsp;уважением, компания MyInsurance',
    },
    approved: {
        message:
            'Мы&nbsp;рады сообщить, что Ваша заявка на&nbsp;&laquo;Страхование&nbsp;квартиры&raquo; была рассмотрена и&nbsp;одобрена. Благодарим за&nbsp;проявленный интерес к&nbsp;нашей компании и&nbsp;надеемся на&nbsp;долгосрочное взаимодействие.',
        requisitesTitle: 'Реквизиты для оплаты:',
        requisites: [
            {
                label: 'Название компании',
                value: 'СПАО «My Insurance»',
            },
            {
                label: 'Расчетный счёт',
                value: '40704917073289876529',
            },
            {
                label: 'БИК',
                value: '011111111',
            },
            {
                label: 'ИНН',
                value: '7711111111',
            },
        ],
    },
    rejected: {
        message:
            'Уважаемый Константин Александрович! К&nbsp;сожалению, мы&nbsp;вынуждены сообщить Вам об&nbsp;отказе на&nbsp;заявку &laquo;Страхование дома&raquo;',
        additionalMessage:
            'Если у&nbsp;Вас возникнут дополнительные вопросы или&nbsp;потребности в&nbsp;страховании, пожалуйста, свяжитесь с&nbsp;нами, и&nbsp;мы&nbsp;будем рады помочь Вам.',
    },
};

export const INSURANCE_CASE = [
    {
        id: '1',
        type: 'Общая информация',
    },
    {
        id: '2',
        type: 'Имущество',
    },
    {
        id: '3',
        type: 'Здоровье',
    },
    {
        id: '4',
        type: 'Путешествия',
    },
    {
        id: '5',
        type: 'Автострахование',
    },
];
