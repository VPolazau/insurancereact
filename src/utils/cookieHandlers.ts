export const setCookie = (name: string, value: string) => {
    const updatedCookie = `${name}=${value}`;
    document.cookie = `${updatedCookie}`;
};

export const getCookie = (name: string) => {
    const matches = document.cookie.match(
        new RegExp(`(?:^|; )${name.replace(/([.$?*|{}()[\]\\/+^])/g, '\\$1')}=([^;]*)`)
    );
    return matches ? matches[1] : undefined;
};
