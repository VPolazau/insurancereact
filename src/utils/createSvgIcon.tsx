import React from 'react';

import { SvgIcon } from '../components/SvgIcon';

export interface ISvgIconProps extends React.SVGProps<SVGSVGElement> {
    titleAccess?: string;
}
export const createSvgIcon = (content: React.ReactNode, viewBox?: string): React.FC<ISvgIconProps> => {
    return function (props) {
        return (
            <SvgIcon viewBox={viewBox} {...props}>
                {content}
            </SvgIcon>
        );
    };
};
