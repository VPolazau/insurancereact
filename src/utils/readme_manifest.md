Должны быть такие разрешения для PWA icons 512/152/144/120/114/76/72/60/57