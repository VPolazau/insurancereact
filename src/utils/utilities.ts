export function decodeHtmlEntities(input: string): string {
    const parser = new DOMParser();
    const decodedString = parser.parseFromString(`<!DOCTYPE html><body>${input}`, 'text/html').body.textContent;
    return decodedString || '';
}
