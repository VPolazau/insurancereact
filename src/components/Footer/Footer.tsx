import React from 'react';
import { useNavigate } from 'react-router-dom';

import { Logo } from '../../assets/icons';
import { COPYRIGHT, MAIN_OFFICE_ADDRESS, PHONE_NUMBER } from '../../utils/constants';
import { Typography } from '../Typography';
import { decodeHtmlEntities } from '../../utils/utilities';

import styles from './styles.module.css';

export const Footer: React.FC = () => {
    const navigate = useNavigate();

    const handleLogoClick = () => {
        navigate('/');
    };

    return (
        <footer className={styles.footer}>
            <div className={styles.container}>
                <div className={styles.logoContainer}>
                    <button className={styles.logoButton} onClick={handleLogoClick} data-testid="iconButtonComponent">
                        <Logo viewBox="0 0 150 46" className={styles.logo} />
                    </button>
                    <Typography className={styles.subtext} variant="small">
                        {decodeHtmlEntities(COPYRIGHT)}
                    </Typography>
                </div>
                <div className={styles.infoContainer}>
                    <div className={styles.textContainer}>
                        <Typography variant="h4">Главный офис</Typography>
                        <Typography className={styles.subtext} variant="small">
                            {decodeHtmlEntities(MAIN_OFFICE_ADDRESS)}
                        </Typography>
                    </div>
                    <div className={styles.textContainer}>
                        <Typography variant="h4">Связаться с нами</Typography>
                        <Typography className={styles.subtext} variant="small">
                            {PHONE_NUMBER}
                        </Typography>
                    </div>
                </div>
            </div>
        </footer>
    );
};
