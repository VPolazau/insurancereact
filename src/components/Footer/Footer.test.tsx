import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { MAIN_OFFICE_ADDRESS, PHONE_NUMBER } from '../../utils/constants';
import { decodeHtmlEntities } from '../../utils/utilities';

import { Footer } from './Footer';

describe('Footer Component', () => {
    it('should render the iconButton', () => {
        render(
            <MemoryRouter>
                <Footer />
            </MemoryRouter>
        );
        expect(screen.getByTestId('iconButtonComponent')).toBeInTheDocument();
    });

    it('should render the main office information', () => {
        render(
            <MemoryRouter>
                <Footer />
            </MemoryRouter>
        );
        expect(screen.getByText('Главный офис')).toBeInTheDocument();
        expect(
            screen.getByText((content, element) => {
                const decodedAddress = decodeHtmlEntities(MAIN_OFFICE_ADDRESS);
                return element?.textContent === decodedAddress;
            })
        ).toBeInTheDocument();
    });

    it('should render the contact information', () => {
        render(
            <MemoryRouter>
                <Footer />
            </MemoryRouter>
        );
        expect(screen.getByText('Связаться с нами')).toBeInTheDocument();
        expect(screen.getByText(PHONE_NUMBER)).toBeInTheDocument();
    });

    it('should match the snapshot', () => {
        const { asFragment } = render(
            <MemoryRouter>
                <Footer />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
