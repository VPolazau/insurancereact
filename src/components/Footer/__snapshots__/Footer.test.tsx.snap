// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should match the snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="container"
    >
      <div
        class="logoContainer"
      >
        <button
          class="logoButton"
          data-testid="iconButtonComponent"
        >
          <svg
            aria-hidden="true"
            class="upIconSvg logo"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 150 46"
          >
            <path
              d="M13.513 29V17.015H14.686L19.293 25.668L23.866 17.015H25.039V29H23.747V19.633L19.752 27.147H18.834L14.788 19.65V29H13.513ZM30.912 29V23.322L26.322 17.015H27.954L31.643 22.098L35.315 17.015H36.896L32.306 23.356V29H30.912Z"
              fill="#22C062"
            />
            <rect
              fill="none"
              height="41"
              rx="3.5"
              stroke="#22C062"
              width="44"
              x="2.5"
              y="2.5"
            />
            <path
              d="M52.479 29V17.015H53.873V29H52.479ZM56.6416 29V20.738H57.9846V22.149C58.2566 21.6277 58.6419 21.2367 59.1406 20.976C59.6393 20.7153 60.2003 20.585 60.8236 20.585C62.7843 20.585 63.7646 21.673 63.7646 23.849V29H62.3876V23.934C62.3876 23.1633 62.2346 22.6023 61.9286 22.251C61.6339 21.8883 61.1579 21.707 60.5006 21.707C59.7413 21.707 59.1349 21.945 58.6816 22.421C58.2396 22.8857 58.0186 23.509 58.0186 24.291V29H56.6416ZM69.0917 29.153C68.4457 29.153 67.8337 29.0623 67.2557 28.881C66.6777 28.6997 66.2017 28.4447 65.8277 28.116L66.2697 27.147C66.6777 27.4757 67.1254 27.7193 67.6127 27.878C68.1114 28.0253 68.6157 28.099 69.1257 28.099C69.7717 28.099 70.2591 27.98 70.5877 27.742C70.9164 27.504 71.0807 27.181 71.0807 26.773C71.0807 26.4557 70.9731 26.2063 70.7577 26.025C70.5424 25.8323 70.2024 25.685 69.7377 25.583L68.1907 25.243C66.7854 24.937 66.0827 24.2117 66.0827 23.067C66.0827 22.319 66.3774 21.7183 66.9667 21.265C67.5674 20.8117 68.3494 20.585 69.3127 20.585C69.8794 20.585 70.4177 20.6757 70.9277 20.857C71.4491 21.027 71.8797 21.282 72.2197 21.622L71.7607 22.591C71.4321 22.285 71.0467 22.0527 70.6047 21.894C70.1741 21.724 69.7434 21.639 69.3127 21.639C68.6781 21.639 68.1964 21.7637 67.8677 22.013C67.5391 22.2623 67.3747 22.591 67.3747 22.999C67.3747 23.6337 67.7941 24.0417 68.6327 24.223L70.1797 24.546C70.9051 24.7047 71.4547 24.9597 71.8287 25.311C72.2027 25.651 72.3897 26.1157 72.3897 26.705C72.3897 27.4643 72.0894 28.065 71.4887 28.507C70.8881 28.9377 70.0891 29.153 69.0917 29.153ZM77.4123 29.153C75.429 29.153 74.4373 28.0593 74.4373 25.872V20.738H75.8143V25.855C75.8143 26.5917 75.9616 27.1357 76.2563 27.487C76.5623 27.8383 77.0383 28.014 77.6843 28.014C78.387 28.014 78.9593 27.7817 79.4013 27.317C79.8433 26.841 80.0643 26.212 80.0643 25.43V20.738H81.4413V29H80.0983V27.606C79.8376 28.1047 79.4693 28.49 78.9933 28.762C78.5286 29.0227 78.0016 29.153 77.4123 29.153ZM84.0176 29V20.738H85.3606V22.217C85.8026 21.2197 86.7093 20.67 88.0806 20.568L88.5736 20.517L88.6756 21.707L87.8086 21.809C87.0266 21.877 86.4316 22.1263 86.0236 22.557C85.6156 22.9763 85.4116 23.5543 85.4116 24.291V29H84.0176ZM92.6284 29.153C92.0617 29.153 91.5517 29.0453 91.0984 28.83C90.6564 28.6033 90.305 28.2973 90.0444 27.912C89.7837 27.5267 89.6534 27.096 89.6534 26.62C89.6534 26.0193 89.8064 25.5433 90.1124 25.192C90.4297 24.8407 90.9454 24.5913 91.6594 24.444C92.3847 24.2853 93.3707 24.206 94.6174 24.206H95.1614V23.679C95.1614 22.9763 95.014 22.472 94.7194 22.166C94.436 21.8487 93.977 21.69 93.3424 21.69C92.8437 21.69 92.362 21.7637 91.8974 21.911C91.4327 22.047 90.9567 22.2737 90.4694 22.591L89.9934 21.588C90.424 21.282 90.9454 21.0383 91.5574 20.857C92.1807 20.6757 92.7757 20.585 93.3424 20.585C94.4077 20.585 95.1954 20.8457 95.7054 21.367C96.2267 21.8883 96.4874 22.6987 96.4874 23.798V29H95.1954V27.572C94.98 28.0593 94.6457 28.4447 94.1924 28.728C93.7504 29.0113 93.229 29.153 92.6284 29.153ZM92.8494 28.116C93.5294 28.116 94.0847 27.8837 94.5154 27.419C94.946 26.943 95.1614 26.3423 95.1614 25.617V25.107H94.6344C93.7164 25.107 92.991 25.1523 92.4584 25.243C91.937 25.3223 91.5687 25.4697 91.3534 25.685C91.1494 25.889 91.0474 26.1723 91.0474 26.535C91.0474 26.9997 91.206 27.3793 91.5234 27.674C91.852 27.9687 92.294 28.116 92.8494 28.116ZM99.042 29V20.738H100.385V22.149C100.657 21.6277 101.042 21.2367 101.541 20.976C102.04 20.7153 102.601 20.585 103.224 20.585C105.185 20.585 106.165 21.673 106.165 23.849V29H104.788V23.934C104.788 23.1633 104.635 22.6023 104.329 22.251C104.034 21.8883 103.558 21.707 102.901 21.707C102.142 21.707 101.535 21.945 101.082 22.421C100.64 22.8857 100.419 23.509 100.419 24.291V29H99.042ZM112.206 29.153C111.379 29.153 110.665 28.9773 110.064 28.626C109.463 28.2633 108.999 27.759 108.67 27.113C108.353 26.4557 108.194 25.6907 108.194 24.818C108.194 23.5033 108.551 22.472 109.265 21.724C109.979 20.9647 110.959 20.585 112.206 20.585C112.716 20.585 113.22 20.6757 113.719 20.857C114.218 21.0383 114.631 21.2933 114.96 21.622L114.484 22.625C114.155 22.3077 113.793 22.0753 113.396 21.928C113.011 21.7807 112.642 21.707 112.291 21.707C111.43 21.707 110.767 21.979 110.302 22.523C109.837 23.0557 109.605 23.8263 109.605 24.835C109.605 25.821 109.837 26.603 110.302 27.181C110.767 27.7477 111.43 28.031 112.291 28.031C112.642 28.031 113.011 27.9573 113.396 27.81C113.793 27.6627 114.155 27.4247 114.484 27.096L114.96 28.099C114.631 28.4277 114.212 28.6883 113.702 28.881C113.203 29.0623 112.705 29.153 112.206 29.153ZM120.13 29.153C118.827 29.153 117.796 28.7733 117.036 28.014C116.277 27.2433 115.897 26.2007 115.897 24.886C115.897 24.036 116.067 23.288 116.407 22.642C116.747 21.9847 117.212 21.4803 117.801 21.129C118.402 20.7663 119.093 20.585 119.875 20.585C120.997 20.585 121.876 20.9477 122.51 21.673C123.145 22.387 123.462 23.373 123.462 24.631V25.158H117.24C117.286 26.0987 117.558 26.8183 118.056 27.317C118.555 27.8043 119.246 28.048 120.13 28.048C120.629 28.048 121.105 27.9743 121.558 27.827C122.012 27.6683 122.442 27.4133 122.85 27.062L123.326 28.031C122.952 28.3823 122.476 28.66 121.898 28.864C121.32 29.0567 120.731 29.153 120.13 29.153ZM119.909 21.605C119.127 21.605 118.51 21.8487 118.056 22.336C117.603 22.8233 117.337 23.4637 117.257 24.257H122.272C122.238 23.4183 122.017 22.7667 121.609 22.302C121.213 21.8373 120.646 21.605 119.909 21.605Z"
              fill="#1F1D1D"
            />
          </svg>
        </button>
        <div
          class="text small subtext"
          data-testid="typographyComponent"
        >
          © 2023  «My Insurance»
        </div>
      </div>
      <div
        class="infoContainer"
      >
        <div
          class="textContainer"
        >
          <div
            class="text h4"
            data-testid="typographyComponent"
          >
            Главный офис
          </div>
          <div
            class="text small subtext"
            data-testid="typographyComponent"
          >
            г. Москва. Одесская улица 10, 5 этаж, офис 13
          </div>
        </div>
        <div
          class="textContainer"
        >
          <div
            class="text h4"
            data-testid="typographyComponent"
          >
            Связаться с нами
          </div>
          <div
            class="text small subtext"
            data-testid="typographyComponent"
          >
            +7 (495) 177 - 37 - 37
          </div>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
